lib/pub_sub/subscriber.rb

```diff
@@ -66,5 +66,19 @@ def queue_arn(region)
     def queue_url(region)
       sqs(region).create_queue(queue_name: PubSub.service_identifier).queue_url
     end
+
+    # Collects all :topics or :subscriptions for a given SNS client
+    def self.collect_all(client, collection_type)
```

Comment:
It is a bit out of place in terms of matching the signature of other APIs. At the same time, it is semantically close to the rest of the class. I considered different options for its placement and thought this would be the cleanest of them all.